src/main/java/dk/cngroup/lentils/service/HintTakenService.java

```diff
@@ -1,31 +1,40 @@
 package dk.cngroup.lentils.service;
 
+import dk.cngroup.lentils.entity.Cypher;
 import dk.cngroup.lentils.entity.Hint;
 import dk.cngroup.lentils.entity.HintTaken;
-import dk.cngroup.lentils.entity.HintTakenKey;
 import dk.cngroup.lentils.entity.Team;
 import dk.cngroup.lentils.repository.HintTakenRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
 import java.util.List;
 
 @Service
 public class HintTakenService {
 
-    HintTakenRepository hintTakenRepository;
+    private HintTakenRepository hintTakenRepository;
 
     @Autowired
     public HintTakenService(HintTakenRepository hintTakenRepository) {
         this.hintTakenRepository = hintTakenRepository;
     }
 
-    public void writeHintTaken(Team team, Hint hint) {
-        HintTakenKey hintTakenKey = new HintTakenKey(team.getId(), hint.getHintId());
-        hintTakenRepository.save(new HintTaken(hintTakenKey));
+    public void takeHint(Team team, Hint hint) {
+        hintTakenRepository.save(new HintTaken(team, hint));
     }
 
-    public List<HintTaken> getHintTakenForTeam(Long teamId) {
-        return hintTakenRepository.findByTeamId(teamId);
+    public int getHintScore(Team team, Cypher cypher) {
```

Comment:
Toto je otazka, jestli tu metodu budeme potrebovat a nebo ne. Kdyztak bych se s tebou na to pak podival, jetli by sla udelat nejak jednoduseji.